### PR TITLE
Add ODBC linking instructions

### PR DESCRIPTION
**Pull request by andralex**
Please let me know of more instructions, thanks.

### CODE REVIEW

**Comment by CyberShadow** [comment]
LGTM

**Comment by andralex** [comment]
@CyberShadow @cruisercoder is the name of unixODBC libunixodbc.a or just libodbc.a?

**Comment by andralex** [comment]
And what would be the pertinent instructions on Win32 and Win64?

**Comment by adamdruppe** [comment]
On Windows, you want to add the odbc32.lib file to the link. I *think* you can just pass it to dmd literally like that, but I know `pragma(lib, "odbc32");` works on Windows (and this is one lib - at least an ancient version of it - that comes with dmd too, yay). Pretty sure it is the same on 64 bit.

**Comment by andralex** [comment]
@adamdruppe could you please give that a brief try? Just paste the example at https://github.com/D-Programming-Language/phobos/pull/3398 in a file and see if it builds with the directives suggested. Thanks!

File: etc/c/odbc/sql.d [hunk 1]
```diff
@@ -10,6 +10,12 @@ original announcement)).
 corresponding to the `sql.h` C header file. It `import`s `public`ly
 `etc.c.odbc.sqltypes` for conformity with the C header.
 
+Note: The ODBC library itself not a part of the `dmd` distribution (and
```
Comment (schuetzm) [inline etc/c/odbc/sql.d]: ... itself _is_ not ...

**Comment by DmitryOlshansky** [comment]
@andralex I have Windows box and here is the result, both `pragma(lib, "odbc32");` and `-Lodbc32.lib` do work. 

However DMD's 32-bit library for ODBC is too old and misses a few symbols, failing to link with
```
 Error 42: Symbol Undefined _SQLSetEnvAttr@16
odbc.obj(odbc) 
 Error 42: Symbol Undefined _SQLAllocHandle@12
--- errorlevel 2
```
Note SqlDrivers was found, but not the other 2 functions.

x86-64 has recent version b/c it uses Microsoft's own SDK and is OK as is.

**Comment by DmitryOlshansky** [comment]
Bug pinned, so I guess just add Windows instructions and we are good to go.
https://issues.dlang.org/show_bug.cgi?id=14787

**Comment by andralex** [comment]
Added Windows instructions, please advise.

**Comment by DmitryOlshansky** [comment]
LGTM

**Comment by andralex** [comment]
@DmitryOlshansky do you have the right to merge (toggle) yet?

**Comment by CyberShadow** [comment]
LGTM

**Comment by CyberShadow** [comment]
Auto-merge toggled on

**Pull request merged**